src/types/data/date.js

```diff
@@ -1,80 +1,168 @@
 import moment from 'moment';
 import React from 'react';
-import {Calendar, TransitionView} from 'react-date-picker';
+import DatetimePicker from 'yet-another-datetime-picker';
 import ImmutablePropTypes from 'react-immutable-proptypes';
-
-import DateField from '~/components/tetheredDateField';
+import TetheredComponent from 'react-tether';
 
 export default function(register) {
   register('date', {
     component: DateComponent,
-    toString: dateToString
+    toString: (value, options) => {
+      return datetimeToString(value, options.get('dateType'));
+    }
   });
 }
 
-const DateComponent = ({name, value, options, disabled, onChange, onBlur}) => {
-  return <DateField
-    className='form-data-input date'
-    value={formatDateValue(value)}
-    placeholder={options.get('placeholder')}
-    dateFormat={getDateFormat(options.get('dateType'))}
-    disabled={disabled}
-    updateOnDateClick={true}
-    onChange={e => onChange(valueToDate(e))}
-    onBlur={onBlur}
-  >
-    <TransitionView cancelButton={false} clearButton={false}>
-      <Calendar />
-    </TransitionView>
-  </DateField>;
-};
+class DateComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.blurTimeout = 0;
+    this.state = createInitialState(props);
```

Comment:
`createInitialState` should be a member of the class so that it is clear that no other functions are using it / it makes no sense outside of the context of this class.